app/kumactl/cmd/apply/apply_test.go

```diff
@@ -294,6 +294,60 @@ var _ = Describe("kumactl apply", func() {
 		Expect(resource.Meta.GetMesh()).To(Equal("meshinit"))
 	})
 
+	It("should apply multiple resources of same type", func() {
+		// given
+		rootCmd.SetArgs([]string{
+			"--config-file", filepath.Join("..", "testdata", "sample-kumactl.config.yaml"),
+			"apply", "-f", filepath.Join("testdata", "apply-multiple-resource-same-type.yaml")},
+		)
+
+		// when
+		err := rootCmd.Execute()
+		// then
+		Expect(err).ToNot(HaveOccurred())
+
+		// when
+		resource := mesh.DataplaneResourceList{}
+		err = store.List(context.Background(), &resource, core_store.ListByMesh("default"))
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		Expect(resource.Items[0].Meta.GetName()).To(Equal("sample1"))
+		Expect(resource.Items[1].Meta.GetName()).To(Equal("sample2"))
```

Comment:
I prefer to have a single given, when, then in the test if possible. If the test has too many `when`, `then` I stop reading those markers. The test overall should be
```
given <- prepare data or environment for test
when <- the test action
then <- the test result
```
always think about what is the action and what is the result of the test. In this case
data are yamls
test action is executing kumactl apply
result is testing if resources are applied.

Therefore I'd mark it like this
```
// given
rootCmd.SetArgs([]string{
    "--config-file", filepath.Join("..", "testdata", "sample-kumactl.config.yaml"),
    "apply", "-f", filepath.Join("testdata", "apply-multiple-resource-same-type.yaml")},
)

// when
err := rootCmd.Execute()

// then
Expect(err).ToNot(HaveOccurred())

resource := mesh.DataplaneResourceList{}
err = store.List(context.Background(), &resource, core_store.ListByMesh("default"))
Expect(err).ToNot(HaveOccurred())
Expect(resource.Items[0].Meta.GetName()).To(Equal("sample1"))
Expect(resource.Items[1].Meta.GetName()).To(Equal("sample2"))
```

If `then` of the test is long (you are checking many things) you can always separate them with `// and`. Like this
```
// then
Expect(err).ToNot(HaveOccurred())

// and 
resource := mesh.DataplaneResourceList{}
err = store.List(context.Background(), &resource, core_store.ListByMesh("default"))
Expect(err).ToNot(HaveOccurred())
Expect(resource.Items[0].Meta.GetName()).To(Equal("sample1"))
Expect(resource.Items[1].Meta.GetName()).To(Equal("sample2"))
```

I sometimes also add a comment to markers like this
```
// then
Expect(err).ToNot(HaveOccurred())

// and resources are saved to the store
resource := mesh.DataplaneResourceList{}
err = store.List(context.Background(), &resource, core_store.ListByMesh("default"))
Expect(err).ToNot(HaveOccurred())
Expect(resource.Items[0].Meta.GetName()).To(Equal("sample1"))
Expect(resource.Items[1].Meta.GetName()).To(Equal("sample2"))
```
for even better readability

don't worry if you are executing `Expect` in given because sometimes it makes sense.

Reply:
thanks so much. noted it :)